app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -114,6 +145,57 @@ class Meta:
             "treatment_branches",
         )
 
+    def _validate_feature_value_against_schema(self, schema, value):
+        try:
+            json_value = json.loads(value)
+        except json.JSONDecodeError as exc:
+            return [exc.msg]
+        try:
+            jsonschema.validate(json_value, schema)
+        except jsonschema.ValidationError as exc:
+            return [exc.message]
+
+    def validate(self, data):
+        data = super().validate(data)
+        # Determine if we require a feature_config
+        feature_config_required = any(
+            [data["reference_branch"].get("feature_enabled", False)]
+            + list(
+                map(lambda b: b.get("feature_enabled", False), data["treatment_branches"])
+            )
+        )
```

Comment:
For things where there's multiple complex nested expressions like this, I usually just break it into separate lines since it usually winds up being the same number of lines and is a little less fragile, something like:

```py
feature_config_required = data["reference_branch"].get("feature_enabled", False)
for branch in data["treatment_branches"]:
    branch_required = branch.get("feature_enabled", False)
    feature_config_requried = feature_config_required or branch_required
```

but that's just a style suggestion